ocs_ci/ocs/utils.py

```diff
@@ -1023,6 +1023,11 @@ def collect_ocs_logs(dir_name, ocp=True, ocs=True, mcg=False, status_failure=Tru
             raise ValueError(
                 f"must-gather fails in an disconnected environment bz-1974959\n{mg_output}"
             )
+        package_log_dir_path = os.path.join(log_dir_path, "rpm_package")
+        try:
+            collect_pod_container_rpm_package(package_log_dir_path)
```

Comment:
This will collect those RPMs for every must gather execution. Even it will try when we have OCP deployment and fail, where OCS deployment is not finished yet.

Also this will collect for every test failure - do you know how long does this takes?
And do we really need to run it for each test failure again and again?

My guess is that we need to collect it only once per whole run? Right?
For production runs we are using collect logs on success parameter and this case I think will be better to do it only once at the end of the execution.

So what I suggest is to remove the call from here, And move it to here:
https://github.com/red-hat-storage/ocs-ci/blob/6c232602d0bf5745f67c28dbcaaadba27c535eb6/tests/conftest.py#L3862 .

WDYT?

This means that it will be called only once at the end of the production run.

Reply:
Updated with changes